spec/requests/users_spec.rb

```diff
@@ -21,4 +21,41 @@
       }))
     end
   end
+
+  it 'when shown, returns a contactable field' do
+    reader_user.contactable = 'consented'
+    reader_user.save!
+    get "/user_accounts/#{reader_user.id}", **api_headers(reader_token)
+    expect_success
+    expect(api_data).to include(contactable: 'consented')
+  end
+
+  it 'accepts a contactable field' do
+    body = {
+      user: {
+        contactable: 'consented'
+      }
+    }
+    patch "/user_accounts/#{reader_user.id}", params: body, **api_with_body_headers(reader_token)
+    expect_success
+    expect(reader_user.reload.contactable_consented?).to be true
+  end
+
+  it 'rejects an invalid contactable field' do
+    body = {
+      user: {
+        contactable: 'cucumber'
+      }
+    }
+    patch "/user_accounts/#{reader_user.id}", params: body, **api_with_body_headers(reader_token)
+    expect(response).to have_http_status(:unprocessable_content)
```

Comment:
you should assert the correct error is returned.

See https://github.com/QutEcoacoustics/baw-server/blob/bdf451e224e8fe0e160d7504828f7f2ef4a98560/spec/requests/analysis_jobs/invalid_requests_spec.rb#L28-L29 for an example